### PR TITLE
parser: better error message #133

### PR DESCRIPTION
**Pull request by lnhote**
### What problem does this PR solve? 
Make parser error/warn message useful.
see https://github.com/pingcap/parser/issues/86

### What is changed and how it works?
1. changed Errorf function, add more context.
2. add/change testcases.
see https://github.com/pingcap/parser/pull/133/files

### Check List 

Tests 

 - Unit test

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/8950)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -34,6 +34,7 @@ require (
 	github.com/grpc-ecosystem/grpc-gateway v1.5.1 // indirect
 	github.com/jonboulle/clockwork v0.1.0 // indirect
 	github.com/klauspost/cpuid v0.0.0-20170728055534-ae7887de9fa5
+	github.com/lnhote/parser v0.0.0-20190103162657-e55d710ca751
```
Comment (zz-jason) [inline go.mod]: We should use "pingcap/parser" directly.

**Comment by zz-jason** [comment]
Looks like we should merge https://github.com/pingcap/parser/pull/133 first, then use the latest "pingcap/parser".

**Comment by lysu** [comment]
LGTM

**Comment by lysu** [comment]
@tiancaiamao @jackysp PTAL

**Pull request merged**